docs/administration/recent_activity/recent_activity.md

```diff
@@ -0,0 +1,231 @@
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   create_draft   admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+  2024-01-29T14:58:53+00:00   319       “Address Book“       create         admin
+  2024-01-29T14:58:53+00:00   319       “Address Book“       publish        admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   create_draft   admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+  2024-01-29T14:58:53+00:00   320       “HQ“                 create         admin
+  2024-01-29T14:58:53+00:00   320       “HQ“                 publish        admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   create_draft   admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   update         admin
+  2024-01-29T14:58:53+00:00   317       “Foo Company Ltd.“   publish        admin
+ --------------------------- --------- -------------------- -------------- -------
+```
+
+### Adding custom Activity Log entries
+
+!!! caution
+
+    Keep activity logging as light as possible. Do not make database request or heavy computation at logging time. Keep them for activity log list display time.
+
+#### Entry
+
+Your custom features can write into the activity log.
+
+First, inject `Ibexa\Contracts\ActivityLog\ActivityLogServiceInterface` into your PHP class having to log an activity (such as a custom event subscriber, event listener, service, or controller).
+
+In the following example, an event subscriber is subscribing to an event dispatched by a custom feature. This event has the information needed by a log entry (see details after the example).
+
+```php
+[[= include_file('code_samples/recent_activity/src/EventSubscriber/MyFeatureEventSubscriber.php') =]]
+```
+
+`ActivityLogService::build` function returns an `Ibexa\Contracts\ActivityLog\Values\CreateActivityLogStruct` which can then be passed to `ActivityLogService::save`.
+
+`ActivityLogService::build` has three arguments:
+
+* `$className` is the FQCN of the object actually manipulated by the feature. For example `Ibexa\Contracts\Core\Repository\Values\Content\Content::class`
+* `$id` is the ID or identifier of the manipulated object. For example, the Content ID cast to string.
+* `$action` is the identifier of the performed object manipulation. For example, `create`, `update` or `delete`.
+
+The returned `CreateActivityLogStruct` is always related to the currently logged-in user.
+
+If the object you log an activity on can become unavailable (like after a `delete` action), you might want to also log the name the object has at log time to be able to display it even when the object becomes unavailable. To add this name, use `CreateActivityLogStruct::setName` before saving the log entry.
```

Comment:
What a complicated sentence!
@bdunogier Is it more clear said like that?
```suggestion
You might want to also log the name of the logged object to be able to display it even if this object becomes unavailable (like after a `delete` action). To add this name, use `CreateActivityLogStruct::setName` before saving the log entry.
```

Reply:
It is !
One thing you could have do in that case: first state the problem, then the solution. Something like "Most objects may deleted or renamed, and this will affect the activity log. Their name at the time of logging can be stored by calling `setName` on the `CreateActivityLogStruct`."

Makes it easier to split the sentence.

Reply:
@bdunogier 
```suggestion
When an object is deleted, the recent activity log list can't reach it anymore. When an object is renamed, it can become confusing if the new name is used in old log entries. To still be able to display the information, you can also store the name of the object at the time with `CreateActivityLogStruct::setName`.
```

Reply:
```suggestion
Most objects may be deleted or renamed, and this will affect the activity log. Their name at the time of logging can be stored by calling `setName` on the `CreateActivityLogStruct`.
```